lib/spack/spack/cmd/list.py

```diff
@@ -116,6 +117,11 @@ def match(p, f):
 
     return sorted(pkgs, key=lambda s: s.lower())
 
+     # If tags have been specified on the command line, filter by tags
+    if args.tags:
+        packages_with_tags = spack.repo.path.packages_with_tags(*args.tags)
+        results = [x for x in results if x.name in packages_with_tags]
```

Comment:
Since this is after the return statement, it is never executed

Reply:
Yupp! , made necessary changes